hack/update-vendor.sh

```diff
@@ -72,28 +72,54 @@ function ensure_require_replace_directives_for_all_dependencies() {
   # collect 'replace' directives that unconditionally pin versions (old=new@version)
   local replace_filter='(.Old.Version == null) and (.New.Version != null)'
 
-  # Capture local require/replace directives before running any go commands that can modify the go.mod file
+  # Capture local require/replace directives from root and staging modules
+  # before running any go commands that can modify the go.mod file
   local require_json="${local_tmp_dir}/require.json"
   local replace_json="${local_tmp_dir}/replace.json"
```

Comment:
if we're not truncating on first write (with `>` redirect), ensure these files don't exist before calling `collect_require_replace_directives` for the first time